R/dates_secuTrial.R

```diff
@@ -28,21 +29,38 @@ datetimes_secuTrial <- function(x, ...) UseMethod("datetimes_secuTrial", x)
 
 #' @rdname dates_secuTrial
 #' @export
-dates_secuTrial.secuTrialdata <- function(object, ...){
+dates_secuTrial.secuTrialdata <- function(object, ...) {
   if (object$export_options$dated) warning("dates already added")
 
-  x <- object$export_options$data_names
-  names(x) <- NULL
-  x <- x[!x %in% object$export_options$meta_names]
+  table_names <- object$export_options$data_names
+  names(table_names) <- NULL
+  # not for meta tables
+  table_names <- table_names[!table_names %in% object$export_options$meta_names]
   # get language and internationalization dictionary for items table
   dict <- object$export_options$dict_items
 
-  obs <- lapply(x, function(obj){
+  obs <- lapply(table_names, function(obj) {
+
     # find date variables
     it <- object[[object$export_options$meta_names$items]]
     qu <- object[[object$export_options$meta_names$questions]]
+    # if meta data is duplicated then the additional "formtablename"
+    # in the items table creates a problem and is thus removed here
+    if(object$export_options$duplicate_meta) {
```

Comment:
this we've done elsewhere by only merging it and qu if meta data isn't duplicated

Reply:
Where exactly? It was failing without this exception.

Reply:
https://github.com/SwissClinicalTrialOrganisation/secuTrialR/blob/df654368e1aee0016539e1339c79f2127fb2f845/R/labels_secuTrial.R#L39

Reply:
The it table contains the itemtype column and qu does not, so the merge is needed here.